singer_sdk/streams/rest.py

```diff
@@ -92,7 +93,13 @@ def __init__(
         if path:
             self.path = path
         self._http_headers: dict = {}
-        self._requests_session = requests.Session()
+
+        self.connector = HTTPConnector(self.config)
+
+        # Override the connector's auth with the stream's auth
+        self.connector._auth = self.authenticator
+
+        self._requests_session = self.connector._session
```

Comment:
Ditto here: what do you think about a public helper method on the HTTPConnector to `.get_session()`, or a public `Connector.session` attr?